aries_cloudagent/wallet/routes.py

```diff
@@ -496,11 +496,12 @@ async def wallet_set_public_did(request: web.BaseRequest):
     mediation_record = await route_manager.mediation_record_if_id(
         profile=profile, mediation_id=mediation_id, or_default=True
     )
-    routing_keys = None
-    mediator_endpoint = None
-    if mediation_record:
-        routing_keys = mediation_record.routing_keys
-        mediator_endpoint = mediation_record.endpoint
+
+    routing_keys, mediator_endpoint = await route_manager.routing_info(
+        profile,
+        None,
```

Comment:
To clarify, we set the value to None here but the returned mediator_endpoint will be set if there's a base mediator; only if there is no mediator in use would the mediator_endpoint come back as a None and then promote_wallet_public_did will overwrite the none with the default endpoint from our settings. Sound right to you?

Reply:
Yes, that's the case I was trying to describe. I was referring to the line 630 in this very file

Reply:
I think I'd like to see this changed up a bit; the logic would be easier to follow if the `promote_wallet_public_did` just accepted an `endpoint` (rather than a `mediator_endpoint`) and then when we call the `route_manager.routing_info`, we pass in the default endpoint value from settings. Fewer magical `None` values. However, I think that can wait for a different PR. Thanks for the clarification :slightly_smiling_face: